content/docs/setup-and-administration/configuring-app-registrations-for-non-interactive-use/index.md

```diff
@@ -0,0 +1,145 @@
+---
+title: "Configuring App Registrations for Non-Interactive Use"
+description: "Creating App Registrations for RKVST"
+lead: "Creating App Registrations for RKVST"
+date: 2021-06-16T11:12:25+01:00
+lastmod: 2021-06-16T11:12:25+01:00
+draft: false
+images: []
+menu: 
+  docs:
+    parent: "setup-and-administration"
+weight: 12
+toc: true
+---
+
+Non-interactive access to the RKVST platform is managed by creating `App Registrations` either via the Manage RKVST Interface in the UI or by using the App Registrations API directly.
+
+`App Registrations` are Authentication Profiles with a `CLIENT ID` and `SECRET` that can then be used to generate JWT Tokens for authenticating to RKVST API Endpoints.
+
+## Creating an App Registration
+
+To enable non-interactive access to RKVST you **must** create your first `App Registration` in the RKVST UI and **only** as a Root User.
+
+### Using the RKVST UI (Required for First-Time Setup)
+
+1. As a Root User visit the APP REGISTRATIONS tab in the `Manage RKVST` Interface
+2. Click CREATE APP REGISTRATION.
+3. Enter any display name you like.
+a. Optional - using the ADD CUSTOM CLAIM button Add any extra claims you require in your access token.
+4. Click CREATE APP REGISTRATION. The response will include the CLIENTID and SECRET required by the archivist token endpoint.
+
+{{< caution >}}
+**Caution:** You **must** take note of the `SECRET` at this point - it can **not** be viewed again later.
+{{< /caution >}}
+
+5. Now you have created your App Registration, follow the steps further below to [test generating a token](./#getting-a-token-with-your-app-registration) and [ensure you can access the RKVST API](./#testing-your-access).
+
+### Using the App Registrations API
+
+The following assumes you already have at least one `App Registration` configured and that you are comfortable generating tokens and using the RKVST API.
+
+If you do not yet have an App Registration configured please follow [the first-time setup guide](./#creating-an-app-registration) to get started.
+
+1. Define your new Application JSON and save it to a file locally. e.g.
+
+```json
+{
+    "display_name": "TrafficLight101",
+    "custom_claims": {
+      "serial_number": "TL1000000101",
+      "has_cyclist_light": "true"
+    }
+}
+```
+
+2. Generate a token using your pre-existing `App Registration` details
+
+```bash
+RESPONSE=$(curl \
+    https://app.rkvst.io/archivist/iam/v1/appidp/token \
+    --data-urlencode "grant_type=client_credentials" \
+    --data-urlencode "client_id=${CLIENTID}" \
+    --data-urlencode "client_secret=${SECRET}")
+
+TOKEN=$(echo -n $RESPONSE | jq .access_token | tr -d '"')
```

Comment:
Do we state somewhere that installing `jq` is a prerequisite? I know I've not done this in the API Reference examples either.

Reply:
We do not, I just quickly copied one of the older examples we have elsewhere, I would like to improve this in general

Reply:
This is NOT resolved we need a better way of doing this because jq shouldn't need to be a pre-req for getting a token, even if we don't solve it technically we need lines to the effect:

> Your token will be in the access_token field

I actually believe we do this more successfully further down would welcome opinions

Reply:
So I copied that whole RESPONSE=$(curl ...) TOKEN=$(...) thing from the previous docs we had on machine auth. I thought somewhere along the line we decided this kind of explicit script rune was actually the least ambiguous way to express things - all of our developer customers ought to know or trivially figure out what jq is and why we used it in the example. 

I'm not sure I see why jq is any more controversial than curl or tr. 

A different way to go is to include an example response and, as suggested, simply say in words "copy the access_token field". Then we only need the curl rune and we do not need any of the noise to do with RESPONSE and TOKEN variables and their construction. 

Speaking as a developer I found the jq rune helpful and the use of tr annoying. `jq -r .access_token` on its own removes the need for tr

Reply:
Speaking from experience, not everyone has access to `jq` inherently as it is not installed by default on most distributions and many companies do not have it enabled in their own repos/mirrors they maintain - so getting hold of it is not always the easiest

I also wouldn't want to mislead people who don't have experience of what `jq` is and why it is used here (a conversation I had a lot in the past as it was a dependency of script we ran regularly in customer envs in a previous job and not many knew what it was and why) that it is an absolute dependency when really it's just making this an easier/smaller snippet

Genuinely. this is a conversation I've had a lot as for most people when json gets involved they're usually using some kind of higher language than bash that can do this stuff natively like python so you'd be suprised by how many people have never encountered `jq` (I actually hadn't till I needed to use the aforementioned script)

I do also not like `tr` either

 I think the most appropriate way is to forgo the trimming of the token and recommend people use the '.access_token' field as we've both said

We can then say, if we feel we still need to,  'a common way to do this is using jq, for example' which better informs/implies customers of where the hard/soft dependencies start and stop and give the example there so then people who need the helpful guidance can still get it

But I think calling out the use of `jq` here is definitely right because it is not installed in most systems by default and customers may need to install it separately, which may in some cases mean getting approvals for certain packages etc etc and it gets messy and increasingly obstructive, and it's not even an actual dependency at the end of the day - just a convenience and at the very least we don't make that part clear enough I think

Reply:
I believe this is resolved @robinbryce and @henry-jewell do you mind confirming you're happy before resolving this?

I'm not sure I need the steps repeated, we may just be able to refer to the bottom section instead